tutorial/stream-client.md

```diff
@@ -0,0 +1,261 @@
+          const resp = await fetch(url, { method: "POST" });
+          if (resp === undefined || resp.status !== 200)
+            throw new Error("failed to join session");
+
+          const response = await resp.json();
+          if (response === undefined || response.status !== "success")
+            throw new Error(response.error);
+
+          return {
+            id: response.metadata.id,
+            websocket: response.metadata.url,
+            stunServers: response.metadata.stun_servers,
+          };
+        }
+
+        disconnect() {}
+      }
+      var stream;
+      window.onload = () => {
+        const connector = new Connector();
+        stream = new AnboxStream({
+          connector: connector,
+          targetElement: "anbox-stream",
+          enableStats: false,
+          stream: {
+            audio: true,
+            video: true,
+          },
+        });
+        stream.connect();
+      };
+    </script>
+  </body>
+</html>
+```
+
+Next, let's make a copy of the Anbox Stream Gateway server to allow the service to validate it:
+
+    # We use cat here to ensure gateway.crt has the users permissions
+    sudo cat /var/snap/anbox-cloud-appliance/common/gateway/server.crt > gateway.crt
+
+As a final step, download a copy of the Anbox streaming SDK and place it inside `stream-client/static`
+
+    curl -o stream-client/static/anbox-stream-sdk.js \
+        https://raw.githubusercontent.com/canonical/anbox-streaming-sdk/refs/heads/main/js/anbox-stream-sdk.js
+
+## Run the stream client
+
+To run the stream client, let's set up a virtual environment for Python to install necessary dependencies:
+
+    cd stream-client
+    python3 -m venv .venv
+    . .venv/bin/activate
+
+Now, we can use `pip` to install all required dependencies
+
+    pip3 install -r requirements.txt
+
+To run the stream client service application, we need to set the necessary environment variables and execute the service via flask.
+
+When installing and initialising the appliance, we would have configured the private IP address of the machine running the appliance. The gateway API is accessible on the local listen address of the appliance on port `9031/tcp`. So that's the IP address and port we use in the GATEWAY_URL.
+
+    export GATEWAY_URL=https://1.2.3.4:9031
+    export GATEWAY_API_TOKEN=xxx
+    export GATEWAY_SERVER_CERT="$PWD"/gateway.crt
+    FLASK_APP=service.py python3 -m flask --cert=adhoc run -h 1.2.3.4 -p 8080
+
+This will start the service and make it accessible at https://1.2.3.4:8080 and print out a username and password needed to access the site.
+
+To access the stream of the instance we created earlier, you can access `https://1.2.3.4:8080/<session_id>` in your browser. Remember to use the session ID of the Android instance we retrieved earlier in place of `<session_id>` and for credentials, use `anbox` as username and the password that the example printed.
```

Comment:
I would never use this 1.2.3.4. It can be confusing. Better specify that this is the private IP address of the machine explicitly, just like it is explicitly specified that one should use the `<session_id>` retrieved above.

Also, "access" is repeated twice. Better use "open" the second time:

```suggestion
    FLASK_APP=service.py python3 -m flask --cert=adhoc run -h 1.2.3.4 -p 8080

This will start the service and make it accessible at https://1.2.3.4:8080 and print out a username and password needed to access the site.

To access the stream of the instance we created earlier, you can open `https://1.2.3.4:8080/<session_id>` in your browser. Remember to use the session ID of the Android instance we retrieved earlier in place of `<session_id>` and for credentials, use `anbox` as username and the password that the example printed.
```